webofneeds/won-owner-webapp/src/main/webapp/app/selectors/connection-selectors.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Created by fsuda on 05.11.2018.
+ */
+
+import {
+  getOwnedNeedByConnectionUri,
+  getOwnedNeeds,
+  getOwnedPosts,
+  getNeeds,
+} from "./general-selectors.js";
+import { isChatConnection } from "../connection-utils.js";
+import won from "../won-es6.js";
+
+/**
+ * Get the connection for a given connectionUri
+ * @param state to retrieve data from
+ * @param connectionUri to find corresponding connection for
+ */
+export function getOwnedConnectionByUri(state, connectionUri) {
+  let need = getOwnedNeedByConnectionUri(state, connectionUri);
+  return need.getIn(["connections", connectionUri]);
+}
+
+/**
+ * Get all connections stored within your own needs as a map
+ * @returns Immutable.Map with all connections
+ */
+export function getOwnedConnections(state) {
+  const needs = getOwnedNeeds(state); //we only check own needs as these are the only ones who have connections stored
+  const connections = needs && needs.flatMap(need => need.get("connections"));
+  return connections;
+}
+
+/**
+ * Get all the connectionUris storid within the state
+ */
+export function getOwnedConnectionUris(state) {
+  const connections = getOwnedConnections(state);
+  return connections && connections.keySeq().toSet();
+}
+
+export function getChatConnectionsByNeedUri(state, needUri) {
```

Comment:
This function fails because it is called with wrong arguments:

```
TypeError: state.get is not a function
    at getNeeds (general-selectors.js:16)
    at getChatConnectionsByNeedUri (connection-selectors.js:43)
    at Controller.hasOpenOrLoadingChatConnections (connections-overview.js:368)
    at fn (eval at compile (angular.js:15551), <anonymous>:4:683)
    at Scope.$digest (angular.js:18253)
    at Scope.$apply (angular.js:18531)
    at angular.js:20448
    at completeOutstandingRequest (angular.js:6362)
    at angular.js:6642
```